src/mux-analytics.brs

```diff
@@ -487,50 +508,57 @@ function muxAnalytics() as Object
-        if m._viewWatchTime <> Invalid
-          if m._viewStartTimestamp <> Invalid
-            date = m._getDateTime()
-            now = 0# + date.AsSeconds() * 1000.0# + date.GetMilliseconds()
-            m._viewWatchTime = now - m._viewStartTimestamp
-          end if
-        end if
-      end if
+  prototype.pollingIntervalHandler = function(pollingIntervalEvent) as Void
+    if m.video = Invalid then return
 
-      ' set buffering metrics
-      if m.video.state = "buffering"
-        if m._Flag_atLeastOnePlayEventForContent = true
-          if m._viewRebufferDuration <> Invalid
-            m._viewRebufferDuration = m._viewRebufferDuration + (m.pollTimer.duration * 1000)
-            if m._viewWatchTime <> Invalid AND m._viewWatchTime > 0
-              m._viewRebufferPercentage = m._viewRebufferDuration / m._viewWatchTime
-            end if
-          end if
-        end if
-      end if
-      if NOT m.video.position = m._Flag_lastReportedPosition
-        if m.video.position > m._Flag_lastReportedPosition
-          ' playposition has increased. This is a progress update
-          if m.video.state = "playing"
-            if m._contentPlaybackTime <> Invalid
-              m._contentPlaybackTime = m._contentPlaybackTime + ((m.video.position - m._Flag_lastReportedPosition) * 1000)
-            end if
-          end if
-        end if
-        if m.video.state = "playing"
-          m._Flag_lastReportedPosition = m.video.position
-        end if
-      end if
-    end if
+    m._setBufferingMetrics()
+    m._updateContentPlaybackTime()
+
+    m._updateTotalWatchTime()
+    m._updateLastReportedPositionFlag()
   end function
 
   ' ' //////////////////////////////////////////////////////////////
   ' ' INTERNAL METHODS
   ' ' //////////////////////////////////////////////////////////////
 
+  prototype._updateLastReportedPositionFlag = function() as Void
+    if m.video.position = m._Flag_lastReportedPosition then return
+    if m.video.state <> "playing" AND m.video.state <> "buffering" then return
+    m._Flag_lastReportedPosition = m.video.position
+  end function
+
+  prototype._updateContentPlaybackTime = function() as Void
+    if m._Flag_isInAdBreak = true then return
+    if m.video.position <= m._Flag_lastReportedPosition then return
+    if m.video.state <> "playing" then return
+    if m._contentPlaybackTime = Invalid then return
+
+    m._contentPlaybackTime = m._contentPlaybackTime + ((m.video.position - m._Flag_lastReportedPosition) * 1000)
+  end function
+
+  prototype._updateTotalWatchTime = function() as Void
+    if m.video.state = "paused" then return
+    if m._viewWatchTime = Invalid then return
+    if m._viewStartTimestamp = Invalid then return
+    if m._viewTimeToFirstFrame = Invalid then return
+    if m._viewRebufferDuration = Invalid then return
+    if m._contentPlaybackTime = Invalid then return
+
+    m._viewWatchTime = m._viewTimeToFirstFrame + m._viewRebufferDuration + m._contentPlaybackTime
```

Comment:
this should also include seeking time, but I can't quite tell if that will be included in rebufferDuration already.